Typing regression on "useI18n()": spyOn return "never" type

### Reporting a bug?

After upgrade from `9.3.0-beta.27`, the mock of `useI18n()` doesn't work:
```typescript
import * as vueI18n from 'vue-i18n'

vi.spyOn(vueI18n, 'useI18n').mockReturnValue({})
```
throws
`Property 'mockReturnValue' does not exist on type 'never'.`

### Expected behavior

No error :)

### Reproduction

1. Create TypeScript project
2. `npm install vue-i118n`
3. Add a simple test  
    ```typescript
    import { vi } from 'vitest'
    import * as vueI18n from 'vue-i18n'
    vi.spyOn(vueI18n, 'useI18n').mockReturnValue({})
    ````
4. `npm run type-check`

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H
    Memory: 10.91 GB / 31.69 GB
  Binaries:
    Node: 20.0.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.5.0 - ~\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Edge: Spartan (44.22621.2134.0), Chromium (117.0.2045.36)
    Internet Explorer: 11.0.22621.1
```


### Screenshot

_No response_

### Additional context

Seems be a regression in #1309 https://github.com/intlify/vue-i18n-next/commit/7ebb0e4f69658ea76237bc24ff3e9e8b722063ec#diff-90783ef144f3a897c890355ff61bd08a99bbd36764128426b6f899a72e766f2eR803

Reverting line to previous version `NonNullable<Options['locale']>` seems fix the problem.
Like the `test:type` checks are still OK, probably the current typing is too strict? 🤔 

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n-next/discussions)

A repo with minimal configuration
ℹ️ 1 commit == 1 CLI command

```sh
git clone git@github.com:pinguet62/vue-i18n--typing-regression.git
cd vue-i18n--typing-regression
npm ci
npm run type-check
```

```
> vue-i18n--typing-regression@0.0.0 type-check
> vue-tsc --noEmit -p tsconfig.vitest.json --composite false

src/sample.spec.ts:8:38 - error TS2339: Property 'mockReturnValue' does not exist on type 'never'.

8         vi.spyOn(vueI18n, 'useI18n').mockReturnValue({
                                       ~~~~~~~~~~~~~~~


Found 1 error in src/sample.spec.ts:8


Process finished with exit code 2
```

--- a/packages/vue-i18n-core/src/i18n.ts
+++ b/packages/vue-i18n-core/src/i18n.ts
@@ -826,7 +826,7 @@ export function useI18n<
   NonNullable<Options['messages']>,
   NonNullable<Options['datetimeFormats']>,
   NonNullable<Options['numberFormats']>,
-  Options['locale'] extends unknown ? string : Options['locale']
+  NonNullable<Options['locale']>
 >
 
 export function useI18n<

--- a/test-dts/vue-i18n/i18n.test-d.ts
+++ b/test-dts/vue-i18n/i18n.test-d.ts
@@ -10,6 +10,7 @@ import {
   useI18n,
   createI18n
 } from '../../packages/vue-i18n-core/src/i18n'
+import * as vueI18n from '../../packages/vue-i18n-core/src/i18n'
 import { SchemaParams, LocaleParams } from '../../packages/core-base/src'
 import { ResourceSchema, MyDatetimeScehma, MyNumberSchema } from '../schema'
 
@@ -505,3 +506,8 @@ expectType<{ zh: {}; 'ja-JP': { short: {} } }>(
   strictDirectI18n.datetimeFormats.value
 )
 expectType<{ ca: { currency: {} } }>(strictDirectI18n.numberFormats.value)
+
+// allow mocking
+vi.spyOn(vueI18n, 'useI18n').mockReturnValue(<ReturnType<typeof useI18n>>{
+  t: (key: string) => `{{${key}}}`
+})

